File: swagger-codegen/sertone_rest_scala_sdk/src/main/scala/io/swagger/client/api/ApplicationsApi.scala
package io.swagger.client.api

import io.swagger.client.model.Applications
import io.swagger.client.ApiInvoker
import io.swagger.client.ApiException

import com.sun.jersey.multipart.FormDataMultiPart
import com.sun.jersey.multipart.file.FileDataBodyPart

import javax.ws.rs.core.MediaType

import java.io.File
import java.util.Date

import scala.collection.mutable.HashMap

class ApplicationsApi(val defBasePath: String = "http://api.dds.sertone.com/v1",
                        defApiInvoker: ApiInvoker = ApiInvoker) {
  var basePath = defBasePath
  var apiInvoker = defApiInvoker

  def addHeader(key: String, value: String) = apiInvoker.defaultHeaders += key -> value 

  /**
   * getApplicationsByUserId() - Gets all applications owned by given user.
   * 
   * @param userId The userId whom to get owned applications 
   * @return List[Applications]
   */
  def getApplicationsByUserId (userId: String) : Option[List[Applications]] = {
    // create path and map variables
    val path = "/users/{userId}/applications".replaceAll("\\{format\\}","json").replaceAll("\\{" + "userId" + "\\}",apiInvoker.escape(userId))


    val contentTypes = List("application/xml", "application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

        
    
    var postBody: AnyRef = null

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
    }

    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "array", classOf[Applications]).asInstanceOf[List[Applications]])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  /**
   * getApplicationsByUserIdAndAppId() - Gets details of given application ID for this given user.
   * 
   * @param userId The userId where application will be retrieved 
   * @param appEui The applicationId to be retrieved 
   * @return Applications
   */
  def getApplicationsByUserIdAndAppId (userId: String, appEui: String) : Option[Applications] = {
    // create path and map variables
    val path = "/users/{userId}/applications/{appEui}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "userId" + "\\}",apiInvoker.escape(userId))

.replaceAll("\\{" + "appEui" + "\\}",apiInvoker.escape(appEui))


    val contentTypes = List("application/xml", "application/json", "application/json")
    val contentType = contentTypes(0)

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]
    val formParams = new HashMap[String, String]

        
    
    var postBody: AnyRef = null

    if(contentType.startsWith("multipart/form-data")) {
      val mp = new FormDataMultiPart()
      
      postBody = mp
    }
    else {
    }

    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, formParams.toMap, postBody, headerParams.toMap, contentType) match {
        case s: String =>
           Some(ApiInvoker.deserialize(s, "", classOf[Applications]).asInstanceOf[Applications])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

}

File: swagger-codegen/sertone_rest_java_sdk/src/main/java/io/swagger/client/api/ApplicationsApi.java
package io.swagger.client.api;

import com.sun.jersey.api.client.GenericType;

import io.swagger.client.ApiException;
import io.swagger.client.ApiClient;
import io.swagger.client.Configuration;
import io.swagger.client.model.*;
import io.swagger.client.Pair;

import io.swagger.client.model.Applications;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-12-09T07:17:28.646Z")
public class ApplicationsApi {
  private ApiClient apiClient;

  public ApplicationsApi() {
    this(Configuration.getDefaultApiClient());
  }

  public ApplicationsApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  /**
   * getApplicationsByUserId() - Gets all applications owned by given user.
   * 
   * @param userId The userId whom to get owned applications (required)
   * @return List<Applications>
   * @throws ApiException if fails to make API call
   */
  public List<Applications> getApplicationsByUserId(String userId) throws ApiException {
    Object localVarPostBody = null;
    
    // verify the required parameter 'userId' is set
    if (userId == null) {
      throw new ApiException(400, "Missing the required parameter 'userId' when calling getApplicationsByUserId");
    }
    
    // create path and map variables
    String localVarPath = "/users/{userId}/applications".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "userId" + "\\}", apiClient.escapeString(userId.toString()));

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();


    
    
    final String[] localVarAccepts = {
      "application/xml", "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/xml", "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] { "server_token", "client_id" };

    GenericType<List<Applications>> localVarReturnType = new GenericType<List<Applications>>() {};
    return apiClient.invokeAPI(localVarPath, "GET", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
      }
  /**
   * getApplicationsByUserIdAndAppId() - Gets details of given application ID for this given user.
   * 
   * @param userId The userId where application will be retrieved (required)
   * @param appEui The applicationId to be retrieved (required)
   * @return Applications
   * @throws ApiException if fails to make API call
   */
  public Applications getApplicationsByUserIdAndAppId(String userId, String appEui) throws ApiException {
    Object localVarPostBody = null;
    
    // verify the required parameter 'userId' is set
    if (userId == null) {
      throw new ApiException(400, "Missing the required parameter 'userId' when calling getApplicationsByUserIdAndAppId");
    }
    
    // verify the required parameter 'appEui' is set
    if (appEui == null) {
      throw new ApiException(400, "Missing the required parameter 'appEui' when calling getApplicationsByUserIdAndAppId");
    }
    
    // create path and map variables
    String localVarPath = "/users/{userId}/applications/{appEui}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "userId" + "\\}", apiClient.escapeString(userId.toString()))
      .replaceAll("\\{" + "appEui" + "\\}", apiClient.escapeString(appEui.toString()));

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();


    
    
    final String[] localVarAccepts = {
      "application/xml", "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/xml", "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] { "server_token", "client_id" };

    GenericType<Applications> localVarReturnType = new GenericType<Applications>() {};
    return apiClient.invokeAPI(localVarPath, "GET", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
      }
}
